### DISCUSSION TITLE
API Coding Style guide

### DISCUSSION THREAD

**Discussion by helderco** [dagger/dagger#12417] [maintainers]
## What is this?

We should have guidelines for our `core/schema/*.graphqls` files, like we used to in [CUE](https://docs.dagger.io/sdk/cue/102745/coding-style).

This helps with consistency and for following best practices that are not only for the GraphQL API, but also that make it **easier to be consumed by the SDKs**.

## Todo

As a first step we can bring quite a few rules that were discussed during the review of the following PR:
- #4367

There's also the undocumented convention of wrapping deprecated methods in backticks (`` ` ` ``):
- https://github.com/dagger/dagger/pull/3882#discussion_r1025869371

This issue can be used for further discussion.

**Comment by TomChv** [comment]
Related to https://github.com/dagger/dagger/issues/4331 so we can enforce the coding style
